org/mozilla/jss/crypto/CMACAlgorithm.java

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.jss.crypto;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Hashtable;
+
+import org.mozilla.jss.asn1.OBJECT_IDENTIFIER;
+
+/**
+ * Algorithms for performing CMACs. These can be used to create
+ * MessageDigests.
+ */
+public class CMACAlgorithm extends DigestAlgorithm {
+
+    protected CMACAlgorithm(int oidIndex, String name, OBJECT_IDENTIFIER oid,
+                            int outputSize) {
+        super(oidIndex, name, oid, outputSize);
+
+        if (oid != null && oidMap.get(oid) == null) {
+            oidMap.put(oid, this);
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////
+    // OID mapping
+    ///////////////////////////////////////////////////////////////////////
+    private static Hashtable<OBJECT_IDENTIFIER, CMACAlgorithm> oidMap = new Hashtable<>();
+
+    /**
+     * Looks up the CMAC algorithm with the given OID.
+     *
+     * @param oid OID.
+     * @return CMAC algorithm.
+     * @exception NoSuchAlgorithmException If no registered CMAC algorithm
+     *  has the given OID.
+     */
+    public static CMACAlgorithm fromOID(OBJECT_IDENTIFIER oid)
+        throws NoSuchAlgorithmException
+    {
+        Object alg = oidMap.get(oid);
+        if (alg == null) {
+            throw new NoSuchAlgorithmException();
+        } else {
+            return (CMACAlgorithm) alg;
+        }
+    }
+
+    /**
+     * CMAC AES-X.  This is a Message Authentication Code that uses a
+     * symmetric key together with the AES cipher to create a form of
+     * signature.
```

Comment:
Expand this comment to detail why we pass `null` as the OID here:

 1. Nobody has defined OIDs yet for CMAC.
 2. While as @edewata points out, static methods aren't inherited (and thus we don't need `fromOID`), there are usages of `fromOID` and `toOID` (the latter being inherited from Algorithm) in pki-core.
 3. We're using PKCS#11-backed implementations here (not OID-backed implementations like e.g., SHA-1 is), so we don't need an OID either.

This will minimize the amount of work in case we later get OID assignments for CMAC.